packages/environment-check/src/checks/destination.ts

```diff
@@ -0,0 +1,287 @@
+        `http://${destName}.dest${
+            odataVersion === OdataVersion.v2 ? getV2CatalogPath(client) : getV4CatalogPath(client)
+        }`
+};
+
+const catalogMessages = {
+    401: (destination: Destination, odataVersion: OdataVersion): string =>
+        t('error.401', { odataVersion, destination: destination.name }),
+    403: (destination: Destination, odataVersion: OdataVersion): string =>
+        t('error.403', { odataVersion, destination: destination.name })
+};
+
+/**
+ * Check a BAS destination, like catalog service v2 & v4.
+ *
+ * @param destination - Destination from list of all destinations
+ * @param username username
+ * @param password password
+ * @returns messages and destination results
+ */
+export async function checkBASDestination(
+    destination: Destination,
+    username?: string | undefined,
+    password?: string | undefined
+): Promise<{ messages: ResultMessage[]; destinationResults: DestinationResults }> {
+    const logger = getLogger();
+    logger.log(t('checkingDestination', { destination: destination.name }));
+
+    const { messages, result: destinationResults } = await checkCatalogServices(destination, username, password);
+
+    logger.push(...messages);
+    return {
+        messages: logger.getMessages(),
+        destinationResults
+    };
+}
+
+/**
+ * Checks for services from catalog requests.
+ *
+ * @param destination sap detination
+ * @param username username
+ * @param password password
+ * @returns Result messages and results of catalog requests
+ */
+async function checkCatalogServices(
+    destination: Destination,
+    username?: string | undefined,
+    password?: string | undefined
+): Promise<{ messages: ResultMessage[]; result: DestinationResults }> {
+    const messages: ResultMessage[] = [];
+
+    const v2results = await catalogRequest(OdataVersion.v2, destination, username, password);
+    messages.push(...v2results.messages);
+
+    const v4results = await catalogRequest(OdataVersion.v4, destination, username, password);
+    messages.push(...v4results.messages);
+
+    const html5DynamicDestination = !!destination.basProperties?.html5DynamicDestination;
+
+    if (!html5DynamicDestination) {
+        messages.push({
+            severity: Severity.Error,
+            text: t('error.missingDynamicDestProperty', { destination: destination.name })
+        });
+    }
+
+    const result: DestinationResults = {
+        v2: v2results.result,
+        v4: v4results.result,
+        HTML5DynamicDestination: html5DynamicDestination
+    };
+
+    return {
+        messages,
+        result
+    };
+}
+
+/**
+ * Performs a catalog request for the given odata version and destination.
+ *
+ * @param odataVersion odataVersion to be used
+ * @param destination destination to be checked
+ * @param username username
+ * @param password password
+ * @returns messages, catalog results, response status
+ */
+async function catalogRequest(
+    odataVersion: OdataVersion,
```

Comment:
That function looks like something that already exists in the `axios-extension` module. Please check the code there and reuse it.